lib/shared_print/finder.rb

```diff
@@ -10,17 +10,25 @@ module SharedPrint
   # * call clusters()    to yield the matching clusters,
   # * call commitments() to yield the matching commitments.
   class Finder
-    attr_reader :organization, :ocn, :local_id, :deprecated, :query
-    def initialize(organization: [], ocn: [], local_id: [], deprecated: false)
+    attr_reader :organization, :ocn, :local_id, :deprecated, :phase, :query
```

Comment:
2 changes:
1) add `phase` as an argument to the constructor, so that it be used in queries
2) add class method that maps phase to committed date (because that's how phases are implemented)